### PR TITLE
Refactor reproducer role

### PR DESCRIPTION
**Pull request by cjeanner**
In order to accept some extension, we first need to refactor things a
bit. This patch is also linked to #766 and #767.

Next steps will include adding devscripts support to the reproducer
role.

As a pull request owner and reviewers, I checked that:
- [X] Appropriate testing is done and actually running

### CODE REVIEW

**Comment by cjeanner** [comment]
@raukadah I'll need that one in before hitting the next steps for devscripts inclusion + the controller-0 vs hypervisor we discussed (tldr; a deploy may not define a controller, meaning the hypervisor (`cifmw_target_host`) will play the controller role).

File: ci_framework/roles/reproducer/tasks/libvirt_layout.yml [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+      block:
+        - name: Get rhos-release
+          ansible.builtin.package:
+            name: "{{ cifmw_repo_setup_rhos_release_rpm }}"
+            state: present
+            disable_gpg_check: true
+
+        - name: Enable RHEL repos
+          ansible.builtin.command:
+            cmd: "rhos-release rhel"
```
Comment (raukadah) [inline ci_framework/roles/reproducer/tasks/libvirt_layout.yml:189]:
```suggestion
            cmd: "rhos-release {{ cifmw_repo_setup_rhos_release_args }}"
```

**Review by raukadah — COMMENTED** [review]
/approve

File: ci_framework/roles/reproducer/tasks/libvirt_layout.yml (hunk 1 above)
Comment (cjeanner) [reply ci_framework/roles/reproducer/tasks/libvirt_layout.yml:189]: hmmmm I don't think we want to get everything OSP on the controller-0, do we?

Comment (raukadah) [reply ci_framework/roles/reproducer/tasks/libvirt_layout.yml:189]: I think `rhos-release rhel` is enough for controller.

**Pull request merged**